src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,47 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
-public class RegistrationServiceImpl implements RegistrationService {
+public class RegistrationServiceImpl
+        implements RegistrationService {
+    private static final int OLDEST_HUMAN_ALIVE = 118;
+    private static final int MIN_AGE = 18;
+    private static final int ZERO_AGE = 0;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User with such login already exists");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password can't be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new InvalidDataException("Password must be at least "
+                    + MIN_PASSWORD_LENGTH + " characters");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidDataException("Age can't be null");
+        }
+        if (user.getAge() < ZERO_AGE) {
```

Comment:
this condition is unnecessary. Users with ages less than 18 don't allow to register.